### PR TITLE
Update hash for caching

### LINKED ISSUE #12402: `cpu (ubuntu-20.04, 3.7, oldest, stable)` fails due to SHA mismatch
## 🐛 Bug
The CI job is failing in master.
```
Collecting numpy>=1.14.6
  Using cached numpy-1.21.5-cp37-cp37m-manylinux_2_12_x86_64.manylinux2010_x86_64.whl (15.7 MB)
ERROR: THESE PACKAGES DO NOT MATCH THE HASHES FROM THE REQUIREMENTS FILE. If you have updated the package versions, please update the hashes. Otherwise, examine the package contents carefully; someone may have tampered with them.
    numpy>=1.14.6 from https://files.pythonhosted.org/packages/50/46/292cff79f5b30151b027400efdb3f740ea03271b600751b6696cf550c10d/numpy-1.21.5-cp37-cp37m-manylinux_2_12_x86_64.manylinux2010_x86_64.whl#sha256=3d893b0871322eaa2f8c7072cdb552d8e2b27645b7875a70833c31e9274d4611 (from scikit-learn>0.22.1->-r requirements/test.txt (line 12)):
        Expected sha256 3d893b0871322eaa2f8c7072cdb552d8e2b27645b7875a70833c31e9274d4611
             Got        447418515[384](https://github.com/PyTorchLightning/pytorch-lightning/runs/5629082072?check_suite_focus=true#step:12:384)153cd6d31e522f98ac6fd9eb4727c1d2dc8b5113f44b24b1ed0a
```
https://github.com/PyTorchLightning/pytorch-lightning/runs/5629082072?check_suite_focus=true

### Environment

- PyTorch Lightning Version (e.g., 1.5.0):
- PyTorch Version (e.g., 1.10):
- Python version (e.g., 3.9):
- OS (e.g., Linux):
- CUDA/cuDNN version:
- GPU models and configuration:
- How you installed PyTorch (`conda`, `pip`, source):
- If compiling from source, the output of `torch.__config__.show()`:
- Any other relevant information:

### Additional context

cc @carmocca @akihironitta @borda

### PR DESCRIPTION
**Pull request by akihironitta**
## What does this PR do?
Fixes #12402.

Adapted from https://github.com/PyTorchLightning/metrics/tree/e82e1496c5d05eb6208d4da2cd0e670ea2a32579/.github/actions/caching

### Does your PR introduce any breaking changes? If yes, please list them.
None

## Before submitting

- [x] Was this **discussed/approved** via a GitHub issue? (not for typos and docs)
- [x] Did you read the [contributor guideline](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/.github/CONTRIBUTING.md), **Pull Request** section?
- [x] Did you make sure your **PR does only one thing**, instead of bundling different changes together?
- [n/a] Did you make sure to **update the documentation** with your changes? (if necessary)
- [n/a] Did you write any **new necessary tests**? (not for typos and docs)
- [n/a] Did you verify new and **existing tests pass** locally with your changes?
- [n/a] Did you list all the **breaking changes** introduced by this pull request?
- [n/a] Did you **update the [CHANGELOG](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/CHANGELOG.md)**? (not for typos, docs, test updates, or internal minor changes/refactorings)

## PR review

Anyone in the community is welcome to review the PR.
Before you start reviewing make sure you have read [Review guidelines](https://github.com/PyTorchLightning/pytorch-lightning/wiki/Review-guidelines). In short, see the following bullet-list:

- [x] Is this pull request ready for review? (if not, please submit in draft mode)
- [x] Check that all items from **Before submitting** are resolved
- [x] Make sure the title is self-explanatory and the description concisely explains the PR
- [x] Add labels and milestones (and optionally projects) to the PR so it can be classified

## Did you have fun?

Make sure you had fun coding 🙃

cc @borda

### CODE REVIEW

File: .github/workflows/ci_test-full.yml [hunk 1]
```diff
@@ -40,8 +40,8 @@ jobs:
       with:
         python-version: ${{ matrix.python-version }}
 
-    - name: Weekly reset caching
-      run: echo "::set-output name=period::$(python -c 'import time ; days = time.time() / 60 / 60 / 24 ; print(int(days / 7))' 2>&1)"
+    - name: Reset caching
+      run: echo "::set-output name=period::$(python -c 'import time ; days = time.time() / 60 / 60 / 24 ; print(int(days / 2))' 2>&1)"
```
Comment (Borda) [inline .github/workflows/ci_test-full.yml:44]:
```suggestion
      run: echo "::set-output name=period::$(python -c 'import time ; days = time.time() / 60 / 60 / 24 ; print(f'd{int(days / 2) * 2}')' 2>&1)"
```
to keep in the name it is a day so it shall be `d2, d4, d6, ...`

**Review by Borda — APPROVED** [review]

**Review by rohitgr7 — APPROVED** [review]

**Review by tchaton — APPROVED** [review]
LGTM !

File: .github/workflows/ci_test-full.yml [hunk 2]
```diff
@@ -40,8 +40,8 @@ jobs:
       with:
         python-version: ${{ matrix.python-version }}
 
-    - name: Weekly reset caching
-      run: echo "::set-output name=period::$(python -c 'import time ; days = time.time() / 60 / 60 / 24 ; print(int(days / 7))' 2>&1)"
+    - name: Reset caching
+      run: echo "::set-output name=period::$(python -c 'import time ; days = time.time() / 60 / 60 / 24 ; print(f'd{int(days / 2) * 2}')' 2>&1)"
```
Comment (rohitgr7) [inline .github/workflows/ci_test-full.yml:44]:
```suggestion
      run: echo "::set-output name=period::$(python -c 'import time ; days = time.time() / 60 / 60 / 24 ; print(f"d{int(days / 2) * 2}")' 2>&1)"
```

File: .github/workflows/ci_test-full.yml [hunk 3]
```diff
@@ -40,8 +40,8 @@ jobs:
       with:
         python-version: ${{ matrix.python-version }}
 
-    - name: Weekly reset caching
-      run: echo "::set-output name=period::$(python -c 'import time ; days = time.time() / 60 / 60 / 24 ; print(int(days / 7))' 2>&1)"
+    - name: Reset caching
+      run: echo "::set-output name=period::$(python -c 'import time ; days = time.time() / 60 / 60 / 24 ; print(f"d{int(days / 2) * 2}")' 2>&1)"
```
Comment (carmocca) [inline .github/workflows/ci_test-full.yml:44]: Why the added `d`?
And why the `* 2`?

File: .github/workflows/ci_test-full.yml [hunk 4]
```diff
@@ -40,8 +40,8 @@ jobs:
       with:
         python-version: ${{ matrix.python-version }}
 
-    - name: Weekly reset caching
```
Comment (carmocca) [inline .github/workflows/ci_test-full.yml:43]: We have this "Weekly reset caching" step in many jobs. Update all?

File: .github/workflows/ci_test-full.yml (hunk 3 above)
Comment (rohitgr7) [reply .github/workflows/ci_test-full.yml:44]: https://github.com/PyTorchLightning/pytorch-lightning/pull/12405#discussion_r831718747

File: .github/workflows/docs-checks.yml [hunk 5]
```diff
@@ -22,19 +22,18 @@ jobs:
...
       # https://github.com/actions/cache/blob/master/examples.md#multiple-oss-in-a-workflow
       - name: Cache pip
         uses: actions/cache@v2
         with:
           path: ~/.cache/pip
-          key: ${{ runner.os }}-pip-td${{ steps.times.outputs.period }}-${{ hashFiles('requirements.txt') }}
+          key: ${{ runner.os }}-pip-td${TIME_PERIOD}-${{ hashFiles('requirements.txt') }}
           restore-keys: |
-            ${{ runner.os }}-pip-td${{ steps.times.outputs.period }}-
+            ${{ runner.os }}-pip-td${TIME_PERIOD}-
```
Comment (carmocca) [inline .github/workflows/docs-checks.yml:36]: Shouldn't this be `{{ env.TIME_PERIOD }}` like the others?

Comment (akihironitta) [reply .github/workflows/docs-checks.yml:36]: Nice catch! Thank you :) I'm throwing a quick PR now :]

Comment (akihironitta) [reply .github/workflows/docs-checks.yml:36]: Updated in #12455.

Comment (Borda) [reply .github/workflows/docs-checks.yml:36]: it does not have any effect but we can do it for consistency

**Pull request merged**